### PR TITLE
Rhode Island Child and Dependent Care Credit

### LINKED ISSUE #2433: Rhode Island Child and Dependent Care Credit
Reference:
https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/2022_1040WE_w_0.pdf
Page 3

https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-01/2021_1040we_w.pdf
Page 3

### PR DESCRIPTION
**Pull request by shuyu7zhan**
Fixes #2433

### Summary
📝📄✅

This pull request implements the Rhode Island Credit for child and dependent care, a state tax credit based on the federal credit. It adds a parameter file, a variable file, two test cases, and a changelog entry for the new feature.

> _`RI_credit` added_
> _A new feature for springtime_
> _Based on `fed_credit`_

### Walkthrough
*  Add Rhode Island Credit for child and dependent care ([link](https://github.com/PolicyEngine/policyengine-us/pull/2442/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
  - Create parameter file for state credit rate ([link](https://github.com/PolicyEngine/policyengine-us/pull/2442/files?diff=unified&w=0#diff-4bae2c0ad6921eebd93c636ec62eaaadf53ddfeb06d62291308aa989038adbefR1-R11))
  - Create variable file for state credit amount ([link](https://github.com/PolicyEngine/policyengine-us/pull/2442/files?diff=unified&w=0#diff-bdaa3417072998ed98d259e82c431d3f22ba501e63838f7fc23794c46c426315R1-R15))
* Add test cases for state credit calculation ([link](https://github.com/PolicyEngine/policyengine-us/pull/2442/files?diff=unified&w=0#diff-c4ba0f29a57254ea7dff88fa92e68eb726a1f32fcedf8ef3143e35f27eeed8ddR1-R15))

### CODE REVIEW

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:1]: need to add check for income

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class ri_cdcc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Rhode Island child and dependent care credit"
+    defined_for = StateCode.RI
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        fed_cdcc_expenses = tax_unit("cdcc_relevant_expenses", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:13]:
```suggestion
        fed_cdcc = tax_unit("cdcc", period)
```

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    label = "Rhode Island child and dependent care credit"
+    defined_for = StateCode.RI
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        ri_income_tax = tax_unit("ri_income_tax", period)
+        fed_cdcc = tax_unit("cdcc", period)
+        rate = parameters(period).gov.states.ri.tax.income.credits.cdcc.rate
+        return min(fed_cdcc * rate, ri_income_tax)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:16]:
```suggestion
        return min_(fed_cdcc * rate, ri_income_tax)
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: run `make format`

File: policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+description: Rhode Island mutiplies the Federal Credit for child and dependent care expenses by this rate.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml:1]:
```suggestion
description: Rhode Island matches the Federal Credit for child and dependent care expenses by this rate.
```

File: policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+values:
+  2021-01-01: 0.25
+metadata:
+  unit: /1
+  label: Rhode Island CDCC match percent
+  reference:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml:7]: need legal code reference

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class ri_cdcc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Rhode Island child and dependent care credit"
+    defined_for = StateCode.RI
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        ri_income_tax = tax_unit("ri_income_tax", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:13]:
```suggestion
```

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        fed_cdcc = tax_unit("cdcc", period)
+        rate = parameters(period).gov.states.ri.tax.income.credits.cdcc.rate
+        return min_(fed_cdcc * rate, ri_income_tax)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:16]:
```suggestion
        return fed_cdcc * rate
```

File: policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    - title: 2022 Form RI-1040 Resident Individual Income Tax Return - page 3
+      href: https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/2022_1040WE_w_0.pdf
+    - title: 2021 Form RI-1040 Resident Individual Income Tax Return - page 3
+      href: https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-01/2021_1040we_w.pdf
+    - title: R.I. Gen. Laws § 44-30-2.6
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ri/tax/income/credits/cdcc/rate.yaml:12]: I don't understand why it restarts from N to A, but this is the second (F)
```suggestion
    - title: R.I. Gen. Laws § 44-30-2.6(F)(I)(g)
```

File: policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+class ri_cdcc(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Rhode Island child and dependent care credit"
+    defined_for = StateCode.RI
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        # ri_income_tax = tax_unit("ri_income_tax", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ri/tax/income/credits/cdcc/ri_cdcc.py:13]:
```suggestion
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Pull request merged**